### PR TITLE
revise pinning nixpkgs

### PR DESCRIPTION
**Pull request by olafklingt**
remove pinning nixpkgs reference
separate pinning nixpkgs from dependency management using niv

### CODE REVIEW

**Comment by olafklingt** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by olafklingt** [comment]
recheck

**Comment by olafklingt** [comment]
After writing ["I keep a note for myself that the “pinning reference” needs some work."](https://discourse.nixos.org/t/pinning-nixpkgs-in-documentation-examples/20749/7) i made this draft today. It is mostly based on nix.dev material written by @domenkozar I would like to know if you agree this PR is going into right direction (as outlined by @fricklerhandwerk ).

**Review by infinisil — CHANGES_REQUESTED** [review]

File: source/tutorials/pinning-nixpkgs.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+## Refering to Nixpkgs 
+
+There are three ways to refer to nixpkgs:
+
+- Setting the environment variable: `$NIX_PATH=nixpkgs=URL` and use the search path `<nixpkgs>` <!-- link to language tutorial -->
+- Using the command line parameter `-I URL` to most of commands like `nix-build`, `nix-shell`, etc <!-- link to ??? tutorial -->
+- Using [builtins.fetchTarball](https://nixos.org/manual/nix/stable/expressions/builtins.html) function that fetches the `URL` at evaluation time <!-- more precise link would be nice -->
+
+Possible `URL` values are:
```
[the remark this answers is not in the document]
Comment (olafklingt) [reply source/tutorials/pinning-nixpkgs.md:39]: We can write `URL` or `path`

File: source/tutorials/pinning-nixpkgs.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+- Local file path. Using just `.` means that nixpkgs is located in current folder.
+- Pinned to a specific commit: `https://github.com/NixOS/nixpkgs/archive/addcb0dddf2b7db505dae5c38fceb691c7ed85f9.tar.gz`
+- Using latest channel, meaning all tests have passed: `http://nixos.org/channels/nixos-21.05/nixexprs.tar.xz`
```
[the remark this answers is not in the document]
Comment (olafklingt) [reply source/tutorials/pinning-nixpkgs.md:43]: thanks for pointing out

[the remark this answers is not in the document]
Comment (olafklingt) [reply source/tutorials/pinning-nixpkgs.md:43]: which are not equivalent? 

commit tar.gz and nixos tar.gz ?

or 

nixos.org and github.com tar.gz that are referring to a release?

**Pull request closed without merging**